policy/aggregation_type.go

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package policy
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/m3db/m3metrics/generated/proto/schema"
+)
+
+// Supported aggregation types.
+const (
+	// Not using iota for better readability.
+	Unknown AggregationType = 0
+	Last    AggregationType = 1
+	Lower   AggregationType = 2
+	Upper   AggregationType = 3
+	Mean    AggregationType = 4
+	Median  AggregationType = 5
+	Count   AggregationType = 6
+	Sum     AggregationType = 7
+	SumSq   AggregationType = 8
+	Stdev   AggregationType = 9
+	P50     AggregationType = 10
+	P95     AggregationType = 11
+	P99     AggregationType = 12
+	P999    AggregationType = 13
+	P9999   AggregationType = 14
+)
+
+const (
+	// TotalAggregationTypes is the number of supported aggregation types.
+	// NB(cw) this should be updated if new AggregationTypes were added.
+	TotalAggregationTypes = 14
+
+	// CompressedSize is the length of the CompressedAggregationTypes.
+	// The CompressedSize will be 1 when totalAggregationTypes <= 63.
+	CompressedSize = TotalAggregationTypes/64 + 1
```

Comment:
probably not for the TotalAggregationTypes,  but since CompressedSize is used in the definition of CompressedAggregationTypes, I think we should export it.